ddtrace/opentelemetry/span_test.go

```diff
@@ -22,38 +22,237 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/httpmem"
 )
 
-func TestSpanMethods(t *testing.T) {
-	testData := struct {
-		env, srv, oldOp, newOp string
-		tags                   [][]string // key - tags[0][0], value - tags[0][1]
-	}{
-		env:   "test_env",
-		srv:   "test_srv",
-		oldOp: "old_op",
-		newOp: "new_op",
-		tags:  [][]string{{"tag_1", "tag_1_val"}, {"opt_tag_1", "opt_tag_1_val"}}}
-	done := make(chan struct{})
-	var payload string
+// helper method to wait until timeout for test agent to receive traces
+func waitForTestAgent(done chan struct{}, timeOut time.Duration, t *testing.T) {
+	select {
+	case <-time.After(timeOut):
+		t.Log("Test agent: timed out waiting for traces")
+		t.FailNow()
+	case <-done:
+		break
+	}
+}
+
+// initializes a test trace provider and agent server
+func getTestTracerProvider(payload *string, done chan struct{},
+	env string, service string, t *testing.T) (*TracerProvider, *http.Server) {
 	s, c := httpmem.ServerAndClient(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
 		case "/v0.4/traces":
+			if h := r.Header.Get("X-Datadog-Trace-Count"); h == "0" {
+				return
+			}
 			buf, err := io.ReadAll(r.Body)
-			if err != nil {
+			if err != nil || len(buf) == 0 {
+				t.Log("Test agent: Error receiving traces")
 				t.Fail()
 			}
-			payload = fmt.Sprintf("%s", buf)
+			*payload = fmt.Sprintf("%s", buf)
```

Comment:
This will produce a beautiful json, that's easier to scan in assert.Contains
```suggestion
			var js bytes.Buffer
			msgp.UnmarshalAsJSON(&js, buf)
			*payload = fmt.Sprintf("%s", js.String())
```